### ISSUE TITLE
windows - access denied error without admin access on packer provisioners

### ISSUE THREAD

**Issue by domoran** [toowoxx/terraform-provider-packer#9]
Short Description: 

when using provisioners under packer and executing the build via terraform, it seems packer uses as a current directory the Windows directory. This leads to an access denied error, when terraform is not invoked with admin access. 

My Environment:
- Terraform v1.1.7 on windows_amd64
- provider registry.terraform.io/toowoxx/packer v0.10.0
- Docker for Window 4.5.1
- packer 1.8.0 

Steps to reproduce: 

- Create a new folder and place the two files main.tf and bug.pkr.hcl inside of it
- Start a command (cmd) shell without elevated access to launch terraform from
- Make sure terraform, packer and docker are on the path
- execute packer to show that the packer build works fine: packer build bug.pkr.hcl
- execute packer via terraform, by executing: terraform apply -auto-approve

Expected behaviour:

both builds succeed, terraform plugin will pass the current work directory to the invoked packer process. 

Observed behaviour: 

Provisioner fails with error message: 
```
Build 'docker.example' errored after 16 seconds 355 milliseconds: Error preparing shell script: open C:\Windows\packer-shell2326743061: Zugriff verweigert
```

which shows that the provisioner tries to place the script command file in the C:\Windows directory which needs elevated rights to write to. 

I attached the relevant files to the issue. 

[main.tf](https://github.com/toowoxx/terraform-provider-packer/files/8499794/main.tf.txt)
[bug.pkr.hcl](https://github.com/toowoxx/terraform-provider-packer/files/8499795/bug.pkr.hcl.txt)
[log.txt](https://github.com/toowoxx/terraform-provider-packer/files/8499796/log.txt)

**Comment by domoran** [comment]
Note: Using the "directory" parameter on the terraform resource to explicitly set the working directory does not help. Using process explorer it seems that the "tmp" environment variable was missing on the packer environment, making the provisioner place temp files inside the SYSTEMROOT directory (C:\Windows). Passing a "TMP=..." variable on the environment works around the problem.  So maybe tmp and path should be passed by default inside the environment of the packer build.

**Comment by simaotwx** [comment]
Thank you for your report.
This seems to be related to https://github.com/toowoxx/terraform-provider-packer/issues/8

Please try the [`keep_environment`](https://registry.terraform.io/providers/toowoxx/packer/latest/docs/resources/image#keep_environment) attribute. If you set it to true, all env variables should get passed through.

**Comment by simaotwx** [comment]
Fixing #8 fixes #9
